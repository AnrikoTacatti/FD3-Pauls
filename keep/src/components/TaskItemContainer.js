"use strict";
import React from 'react';
import { connect } from 'react-redux';
import TaskItem from './TaskItem.js';
import {
    actionopenFormNewItemEdit,
    actionremoveTaskItem,
    actionsetTaskItemStyle,
    actionsetTaskItemPin
} from '../actions/TaskItem.js';


export class TaskItemContainer extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            keychapter: this.props.keychapter,
            keyitem: this.props.keyitem,
            del: false
        }
    }

    openFormNewItemEdit = () => {
        let data = { name: this.props.data.name, text: this.props.data.text, activ: true, keychapter: this.props.keychapter, keyitem: this.props.keyitem };
        actionopenFormNewItemEdit(data, this.props.dispatch);
    }
    deleteTaskItem = () => {
        if (this.state.del === true) {
            let isDelete = window.confirm("Вы действительно хотите удалить?");
            if (isDelete) {
                this.divitem.className = this.divitem.className + " del";
                let data = { keychapter: this.state.keychapter, keyitem: this.state.keyitem };
                actionremoveTaskItem(data, this.props.dispatch);
            }
            else {
                this.setState({ del: false });
            }
        }
    }

    addDel = () => {
        this.setState({ del: true });
    }

    getTime(timestamp) {
        let d = new Date(timestamp);
        let timeStampCon = d.getDate() + '/' + (d.getMonth()) + '/' + d.getFullYear() + " " + d.getHours() + ':' + (d.getMinutes() < 10 ? '0' + d.getMinutes() : d.getMinutes());
        return timeStampCon;
    }

    setColor = (style, e) => {
        let data = { style: style, keychapter: this.state.keychapter, keyitem: this.state.keyitem };
        actionsetTaskItemStyle(data, this.props.dispatch);
    }

    setPin = (ifo) => {
        let pin = true;
        if (ifo === true) pin = false;
        let data = { pin: pin, keychapter: this.state.keychapter, keyitem: this.state.keyitem };
        actionsetTaskItemPin(data, this.props.dispatch);
    }

    render() {
        // console.log("render TaskItemContainer", this.props);
        return (
            <TaskItem {...this.props} del={this.state.del} cbopenFormNewItemEdit={this.openFormNewItemEdit} cbsetPin={this.setPin}
                cbaddDel={this.addDel} cbdeleteTaskItem={this.deleteTaskItem} cbgetTime={this.getTime} cbsetColor={this.setColor} ></TaskItem>
        )
    }
}

export default connect()(TaskItemContainer);
