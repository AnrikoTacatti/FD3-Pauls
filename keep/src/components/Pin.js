"use strict";
import React from 'react';
import FormTaskItem from './FormTaskItem.js';
import TaskItemContainer from './TaskItemContainer.js';

class Pin extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    forTaskListschild = () => {
        let tasklistitem = [];
        var i = 0;
        let listtask = [];
        if (this.props.TaskListsItemsSort.length) listtask = this.props.TaskListsItemsSort;

        let index = 0;
        for (let i = 0; i < this.props.TaskListsItemsSort.length; i++) {
            let keyitem = listtask[i].key;
            let keychapter = listtask[i].keychapter;
            if (listtask[i].pin === true) {
                index++;
                tasklistitem.push(
                    <TaskItemContainer data={listtask[i]} keychapter={keychapter} keyitem={keyitem} key={keyitem} index={index}
                        attrdata={this.props.locationPathname === undefined ? "all" : this.props.locationPathname}
                    />
                )
            }
        }
        if (tasklistitem.length == 0) {
            return "Закрепленных нет"
        }
        return tasklistitem;
    }

    render() {
        console.log("render Pin", this.props);
        return (
            <React.Fragment>
                <div className="task-chapter" >
                    <style>
                        {` .task-item{ -webkit-animation-name: none;
                                animation-name: none;} 
                            .${this.props.locationPathname !== undefined ? this.props.locationPathname : "all"} { 
                                -webkit-animation-name: slideInUp;
                                animation-name: slideInUp;
                                opacity: 0; }
                            `
                        }
                    </style>
                    <div className="task-chapter__header">
                        <div className="task-chapter__title">
                            Закрепленные
                        </div>

                    </div>
                    <div className="task-item-list">
                        {this.forTaskListschild()}
                    </div>
                </div>
                <FormTaskItem />
            </React.Fragment>

        );
    }
}

export default Pin;
