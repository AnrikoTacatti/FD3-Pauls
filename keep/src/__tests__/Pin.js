"use strict";
import React from 'react';
import Pin from '../components/Pin';
import TaskItemContainer from '../components/TaskItemContainer.js';

let taskListsItemsSort = [
    {
        "name": "Заметки учеба №111",
        "pin": true,
        "style": {
            "backgroundColor": "#ffecf1",
            "border": "1px solid #f9c8ca"
        },
        "text": "Заметки учеба №1",
        "time": 1636466420836,
        "key": "-Mnu71_BYp5YokQlvrg4",
        "keychapter": "-MnosmgmdFWwUg_Qyj5o"
    },
    {
        "name": "Заметки хобби № 7",
        "text": "Заметки хобби № 7",
        "time": 1636284129331,
        "key": "-Mnu7_uMv7uxZQNdvVOf",
        "keychapter": "-Mnot8F4fnIev0gGlnxg"
    },
    {
        "name": "Заметки хобби № 1",
        "text": "Заметки хобби № 1",
        "time": 1636284087310,
        "key": "-Mnu7QdYFIdU_yM1_eCE",
        "keychapter": "-Mnot8F4fnIev0gGlnxg"
    },
    {
        "name": "Заметки учеба №11",
        "style": "background-color: rgb(232, 250, 252); border: 1px solid rgb(178, 217, 236);",
        "text": "Заметки учеба №11",
        "time": 1636284065035,
        "key": "-Mnu7FEYa0Vr8GBoy_Mc",
        "keychapter": "-MnosmgmdFWwUg_Qyj5o"
    },
    {
        "name": "Заметки работа №2",
        "pin": true,
        "text": "Заметки работа №2",
        "time": 1636283923175,
        "key": "-Mnu6n_j_6Szfb5fRtuR",
        "keychapter": "1"
    },
    {
        "name": "Заметки работа №1",
        "pin": true,
        "text": "Заметки работа №1",
        "time": 1636283916422,
        "key": "-Mnu6lwJvGFmsBIjhPBh",
        "keychapter": "1"
    },
    {
        "name": "Мои заметки № 10",
        "text": "Мои заметки № 10",
        "time": 1636283900074,
        "key": "-Mnu6hxwC6QMIoX2YcMT",
        "keychapter": "0"
    },
    {
        "name": "Мои заметки № 8",
        "text": "Мои заметки № 8",
        "time": 1636283876009,
        "key": "-Mnu6c5GJtYyhTBYAQ9d",
        "keychapter": "0"
    },

];

describe('Pin-test', () => {
    let state = {
        TaskListsItemsSort: taskListsItemsSort,
        locationPathname: "pin",
    }
    const component = shallow(
        <Pin locationPathname={state.locationPathname} TaskListsItemsSort={state.TaskListsItemsSort} />
    );

    test('Pin Snapshot', () => {
        expect(component).toMatchSnapshot();
    });
    test('Pin task-chapter__title', () => {
        const title = component.find(".task-chapter__title").text();
        expect(title).toBe("Закрепленные");
    });
    test('Search  TaskItemContainer length', () => {
        const TaskItemContainerobj = component.find(TaskItemContainer);
        expect(TaskItemContainerobj).toHaveLength(3);
    });
});



