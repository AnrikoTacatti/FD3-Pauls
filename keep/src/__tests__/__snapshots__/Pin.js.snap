// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pin-test Pin Snapshot 1`] = `
<Fragment>
  <div
    className="task-chapter"
  >
    <style>
       .task-item{ -webkit-animation-name: none;
                                animation-name: none;} 
                            .pin { 
                                -webkit-animation-name: slideInUp;
                                animation-name: slideInUp;
                                opacity: 0; }
                            
    </style>
    <div
      className="task-chapter__header"
    >
      <div
        className="task-chapter__title"
      >
        Закрепленные
      </div>
    </div>
    <div
      className="task-item-list"
    >
      <Connect(TaskItemContainer)
        attrdata="pin"
        data={
          Object {
            "key": "-Mnu71_BYp5YokQlvrg4",
            "keychapter": "-MnosmgmdFWwUg_Qyj5o",
            "name": "Заметки учеба №111",
            "pin": true,
            "style": Object {
              "backgroundColor": "#ffecf1",
              "border": "1px solid #f9c8ca",
            },
            "text": "Заметки учеба №1",
            "time": 1636466420836,
          }
        }
        index={1}
        key="-Mnu71_BYp5YokQlvrg4"
        keychapter="-MnosmgmdFWwUg_Qyj5o"
        keyitem="-Mnu71_BYp5YokQlvrg4"
      />
      <Connect(TaskItemContainer)
        attrdata="pin"
        data={
          Object {
            "key": "-Mnu6n_j_6Szfb5fRtuR",
            "keychapter": "1",
            "name": "Заметки работа №2",
            "pin": true,
            "text": "Заметки работа №2",
            "time": 1636283923175,
          }
        }
        index={2}
        key="-Mnu6n_j_6Szfb5fRtuR"
        keychapter="1"
        keyitem="-Mnu6n_j_6Szfb5fRtuR"
      />
      <Connect(TaskItemContainer)
        attrdata="pin"
        data={
          Object {
            "key": "-Mnu6lwJvGFmsBIjhPBh",
            "keychapter": "1",
            "name": "Заметки работа №1",
            "pin": true,
            "text": "Заметки работа №1",
            "time": 1636283916422,
          }
        }
        index={3}
        key="-Mnu6lwJvGFmsBIjhPBh"
        keychapter="1"
        keyitem="-Mnu6lwJvGFmsBIjhPBh"
      />
    </div>
  </div>
  <Connect(FormTaskItem) />
</Fragment>
`;
